datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx

```diff
@@ -91,18 +89,19 @@ export default function LineageTreeNodeAndEdgeRenderer({
                             strokeWidth="1"
                             markerEnd={`url(#triangle-downstream${isHighlighted ? '-highlighted' : ''})`}
                             markerStart={`url(#triangle-upstream${isHighlighted ? '-highlighted' : ''})`}
-                            data-testid={`edge-${link.source.data.urn}-${link.target.data.urn}-${direction}`}
+                            data-testid={`edge-${link.source.data.urn}-${link.target.data.urn}-${link.target.direction}`}
                         />
                     </Group>
                 );
             })}
-            {nodesToRender.map((node) => {
+            {nodesToRender.map((node, index) => {
                 const isSelected = node.data.urn === selectedEntity?.urn;
                 const isHovered = node.data.urn === hoveredEntity?.urn;
+                const key = `node-${node.data.urn}-${node.direction}-${index}`;
 
                 return (
                     <LineageEntityNode
-                        key={`node-${node.data.urn}-${direction}`}
+                        key={key}
```

Comment:
Not sure if this is relevant, but is there a way to use a triplet to render the node?

node-node.data.urn-dest.data.urn-direction?

Or is this not really the problem

Reply:
yeah so the problem when switching base nodes is we had keys here that weren't unique and we would leave the old base node rendered.

but as far as the general issue of having duplicate nodes and dragging and what not, having some combo like that could certainly work. idk if it's relevant in this specific situation though